iXBRLViewerPlugin/viewer/src/js/textblockviewer.js

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Dialog } from './dialog.js';
+
+export class TextBlockViewerDialog extends Dialog {
+    constructor() {
+        super(".dialog.text-block-viewer");
+        this.addButton("Dismiss", true);
+    }
+
+    htmlWrapString(content) {
+        const str =
+            "<!DOCTYPE html><html><head><title></title></head><body>" 
+            + content 
+            + "</body></html>";
+        return str;
+    }
+
+    displayTextBlock(textBlockValue) {
+        const iframe = this.node.find("iframe").get(0);
+        const doc = iframe.contentDocument || iframe.contentWindow.document;
+        doc.open();
+        const html = this.htmlWrapString(textBlockValue);
+        const parser = new DOMParser();
+        const xml = parser.parseFromString(html, 'text/xml');
+        const errorNode = xml.querySelector("parsererror");
+        if (errorNode) {
+            doc.write(this.htmlWrapString('<div style="color: red">Text block contains invalid XML.</div>'));
```

Comment:
Out of curiosity, in what case can this happen?  A comment explaining (even if it just says it should never happen) would be welcome.

Should we show a warning but still show the "raw" fact value?

Reply:
Now that it's limited to `escape="true"`, it should be impossible.

When I originally wrote it, I was thinking we might use the text block viewer on `escape="false"` facts, as ESEF permit this and it might be useful for people to see what such facts look like, but I get the impression that such facts are not common.

On reflection, if we're not attempting to show `escape="false"` facts, I'm not sure this validation is adding any value, and it's potentially expensive.  Even malformed XML can't escape the iframe, and it's not providing any protection against malicious HTML, so we should probably just remove it.